api/v1alpha1/openapi.yaml

```diff
@@ -3197,6 +3201,26 @@ components:
         - "DeviceUpdatedStatusOutOfDate"
         - "DeviceUpdatedStatusUpdating"
         - "DeviceUpdatedStatusUnknown"
+    DeviceLifecycleStatus:
+      type: object
+      required:
+        - status
+      properties:
+        status:
+          $ref: "#/components/schemas/DeviceLifecycleStatusType"
+          description: "Status type of the device lifecycle."
+        info:
+          type: string
+          description: "Human readable information about the device lifecycle status."
+    DeviceLifecycleStatusType:
+      type: string
+      enum:
+        - "EnrollmentPending"
+        - "Enrolled"
+        - "Decommissioning"
+        - "DecommissionError"
+        - "DecommissionTimeout"
```

Comment:
The last three `DecommissionError`, `DecommissionTimeout`, and `Decommissioned` would be end-states. They are the view of the service, based in part on what is reported by the agent (in a different field). I could make a diagram.

Reply:
@fzdarsky Would you still like the diagram or does this make sense? These are set by the service based on `Condition`s set by the agent (which can be `DecommissionStarted`, `DecommissionComplete`, or `DecommissionError`.)

Reply:
Update, there is one `Condition` called `DeviceDecommissioning` set by the device and `Reasons` that give state updates (also set by the device). See #715

Reply:
> @fzdarsky Would you still like the diagram or does this make sense? These are set by the service based on `Condition`s set by the agent (which can be `DecommissionStarted`, `DecommissionComplete`, or `DecommissionError`.)

We'll need to add that diagram to the user docs, yes, but let's make it a separate PR.

Reply:
> Update, there is one `Condition` called `DeviceDecommissioning` set by the device and `Reasons` that give state updates (also set by the device). See #715

Consequence is this list needs updating to `Enrolled`, `Decommissioning`, and `Decommissioned`.

Reply:
If our lifecycle states are `Enrolled`, `Decommissioning`, and `Decommissioned`, where can we display/expose the information that decommissioning ended in a timeout? The timeout is based on the Condition (updated by the device) + server-side logic, which makes it a good candidate to be displayed here. Our other options are:

- Add a LifecycleSummary field which would have the overall status (Enrolled, Decommissioning, Decommissioned) and include details in the Lifecycle.Status (not summary)
- Include the details like timeout in the `info` field - but this would not be machine-readable or easy to filter on
- Have the service overwrite `Conditions` or `Reasons` - which defeats the purpose of having the separation
- Include another field on DeviceStatus to hold the timeout info - but it seems confusing to not include it with the other decommissioning status info

I would advocate for keeping the timeout in this `lifecycle.status` field, plus or minus adding a Summary status.

cc @hexfusion

Reply:
the status of the agent is reported via condition `Decommissioning` if there was an issue  `Timeout` I would expect Reason=DecommissionStateError Message "Decommissioning failed: timeout". The `lifecycle.status.{info,status}` would reflect.

Reply:
The problem is, with our current plan the timeout is only computed on server side. So the Reason would not be able to reflect the timeout and would still show `DecommissionStateStarted`.

Reply:
Gotcha Timeout then would work the same as `Unknown`[1] when the device doesn't check-in?

[1] https://github.com/flightctl/flightctl/blob/main/internal/tasks/device_disconnected.go#L32

Reply:
Yes, it seems similar to that. We need a field in which to display that, specifically for decommissioning, if we want to have the timeout.

Reply:
I suppose if devices disappear from all views once they reach a `Decommissioned` status, then we don't need to show this, but we have not come to that conclusion.

Reply:
>Why do you assume devices disappear from views?

I'm not assuming it, in fact I think it would be the wrong choice. I am trying to understand whether you or others are assuming it. If not, we need a field to hold this information. As I've thought about it today, I think the addition of a `DeviceLifecycleSummary` with values `Enrolled`, `Decommissioning`, `Decommissioned` and then a separate `DeviceLifecycleStatus` with `DecommissioningStarted`, `DecommissioningTimeout` etc. for more detail would be the best way forward.